functions/snippets/src/test/java/HelloBackgroundIT.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.google.cloud.functions.HttpRequest;
+import com.google.cloud.functions.HttpResponse;
+import com.google.common.testing.TestLogHandler;
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.StringReader;
+import java.io.StringWriter;
+import java.util.Optional;
+import java.util.logging.Logger;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.contrib.java.lang.system.EnvironmentVariables;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest(HelloBackground.class)
+public class HelloBackgroundIT {
+  @Mock private HttpRequest request;
+  @Mock private HttpResponse response;
+
+  private EnvironmentVariables environmentVariables;
+
+  private Logger logger = Logger.getLogger(HelloBackground.class.getName());
+  private TestLogHandler logHandler;
+
+  @Before
+  public void beforeTest() throws Exception {
+    environmentVariables = new EnvironmentVariables();
+
+    logHandler = new TestLogHandler();
+    logger.addHandler(logHandler);
+
+    // Use a new mock for each test
+    request = mock(HttpRequest.class);
```

Comment:
I don't think mocking is a good way to test this functionality. In particular, mocking a third-party interface has consequences and I don't think we should encourage users to do so. 

We should test these (and encourage users to test these) with a nested class that implements the interface instead.